### ISSUE TITLE
Setting `VisualStateGroups` in a style and set the style to a `Button` doesn't work

### ISSUE THREAD

**Issue by AndreasReitberger** [dotnet/maui#16569]
### Description

If I create a `Style`, for instance for a `Button` and set the `VisualStateGroups` in a style with a key, it doesn't work when the control uses the style by it's key.

For instance (see repro)

```xaml
<Style x:Key="DefaultButtonStyle" TargetType="Button">
    <Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource White}, Dark={StaticResource Primary}}" />
    <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource White}}" />
    <Setter Property="FontFamily" Value="OpenSansRegular"/>
    <Setter Property="FontSize" Value="14"/>
    <Setter Property="CornerRadius" Value="8"/>
    <Setter Property="Padding" Value="14,10"/>
    <Setter Property="MinimumHeightRequest" Value="44"/>
    <Setter Property="MinimumWidthRequest" Value="44"/>
    <Setter Property="VisualStateManager.VisualStateGroups">
        <VisualStateGroupList>
            <VisualStateGroup x:Name="CommonStates">
                <VisualState x:Name="Normal" />
                <VisualState x:Name="Disabled">
                    <VisualState.Setters>
                        <Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource Gray950}, Dark={StaticResource Gray200}}" />
                        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource Gray600}}" />
                    </VisualState.Setters>
                </VisualState>
            </VisualStateGroup>
        </VisualStateGroupList>
    </Setter>
</Style>

<Style TargetType="Button" BasedOn="{StaticResource DefaultButtonStyle}" />
```

In the `MainPage.xaml`, I have three buttons. One is enabled, one is disabled using directly the style by its key, and one disabled without setting the style directly.

```xaml
<Button
    x:Name="CounterBtn"
    Text="Click me"
    SemanticProperties.Hint="Counts the number of times you click"
    Clicked="OnCounterClicked"
    HorizontalOptions="Center" />

<Button
    Text="Click me (Disabled)"
    SemanticProperties.Hint="Counts the number of times you click"
    Clicked="OnCounterClicked"
    HorizontalOptions="Center"
    IsEnabled="False"
    Style="{StaticResource DefaultButtonStyle}"
    >
</Button>

<Button
    Text="Click me (Disabled) - Without key"
    SemanticProperties.Hint="Counts the number of times you click"
    Clicked="OnCounterClicked"
    HorizontalOptions="Center"
    IsEnabled="False"
    />
```

The result looks like that:

![image](https://github.com/dotnet/maui/assets/2952443/b1b046f4-b468-4f05-a798-09a740491aaa)

Basically the both disabled buttons actually should look the same, but they doesn't

### Steps to Reproduce

1. See the repro
2. Run it and see that both buttons behave differently

### Link to public reproduction project repository

https://github.com/AndreasReitberger/VisualStateIssueMaui

### Version with bug

7.0.49

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows

### Did you find any workaround?

Setting the style not by key

**Comment by StephaneDelcroix** [comment]
Does it change a thing if you set the Style="..." before the IsEnabled="false"

**Comment by AndreasReitberger** [comment]
> Does it change a thing if you set the Style="..." before the IsEnabled="false"

Yes, indeed. In the repro it does work if you exchange the positions.

```xaml
<Button
    Text="Click me (Disabled)"
    SemanticProperties.Hint="Counts the number of times you click"
    Clicked="OnCounterClicked"
    HorizontalOptions="Center"
    Style="{StaticResource DefaultButtonStyle}"
    IsEnabled="False"
    >
</Button>
```

Same also happens, if the `Command.CanExecute` actually disables the button.

![image](https://github.com/dotnet/maui/assets/2952443/026585af-2eb5-490f-8f58-22bff3514e96)

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 1.0. Repro on Windows 11 .NET 8, Android 13.0-API33 and iOS 16.4 with below Project:
[VisualStateIssueMuai.zip](https://github.com/dotnet/maui/files/12377944/VisualStateIssueMuai.zip)

**Comment by AndreasReitberger** [comment]
Any news?
